src/App.tsx

```diff
@@ -1,27 +1,63 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { get5First, getAll, getRedGoods } from './api/goods';
 
 // import { getAll, get5First, getRed } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  async function shouldShowAllGoods() {
+    const allGoods = await getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    setGoods(allGoods);
+  }
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  async function shouldShowFirstFive() {
+    const fiveFirstGoods = await get5First();
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    setGoods(fiveFirstGoods);
+  }
+
+  async function shouldShowRed() {
+    const redGoods = await getRedGoods();
+
+    setGoods(redGoods);
+  }
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={shouldShowAllGoods}
```

Comment:
```suggestion
        onClick={handleGoods(getAll)}
```
same for other buttons